### DISCUSSION TITLE
When triage rights are granted through a team, the "Convert to discussion" button does not appear

### DISCUSSION THREAD

**Discussion by jarrodmillman** [community/community#2993] [Discussions]
Thanks for Discussions!  It looks like it will be very useful.  This is a bug report (I think) and I wasn't sure where to put it.  Please let me know if this is the wrong place.

We would like members with triage permission to be able to individually convert issues to discussion.  Currently they can not.

On
- https://docs.github.com/en/free-pro-team@latest/discussions/managing-discussions-for-your-community/moderating-discussions

it states
```
Discussions are an open forum for conversation among maintainers and the community for a project on GitHub.
If you have triage permissions for a repository, you can help moderate a project's discussions by marking comments
as answers, locking discussions that are not longer useful or are damaging to the community, and converting issues
to discussions when an idea is still in the early stages of development.
```
My reading is that "if you have triage permissions for a repository", then you can "convert issues to discussions when an idea is still in the early stages of development."  I am not sure what "when an idea is still in the early stages of development" means, so maybe that is the issue.  The issue we are having issues with is
- https://github.com/networkx/networkx/issues/4439

It is only a few days old, but it has had a few comments.  Can you clarify what restriction is meant by "when an idea is still in the early stages of development" ?

On
- https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization

in the "Repository access for each permission level" section the "Individually convert issues to discussions" line has an "X" in the "Triage" column, which again makes me think that someone with triage permissions to the repository should be able to individually convert an issue to a discussion.

On
- https://docs.github.com/en/free-pro-team@latest/discussions/managing-discussions-for-your-community/moderating-discussions#converting-an-issue-to-a-discussion

it says "People with write access to a repository can bulk convert issues based on labels."  That seems different that individually converting an issue to a discussion.

Are individuals with triage permission to a repository supposed to be able to individually convert an issue to a discussion (that is what I would prefer and what I understand is intended by reading the documentation.)

If this is the case, it looks like there is a bug.  @MridulS is a core developer of NetworkX
- https://github.com/orgs/networkx/teams/core-developers/members

and the core-developers of NetworkX have triage permission to the networkx/networkx repository
- https://github.com/orgs/networkx/teams/core-developers/repositories

but he does **not** have a "Convert to discussion" option on the right hand side of the issue under the list of participants:
- https://github.com/networkx/networkx/issues/4439

**Comment by becca** [comment]
Hello there and thank you @jarrodmillman for the really well documented request/feedback (and for looking through the docs first!) 

Those with `triage` permissions **should** indeed be able to convert individual issues into Discussions. I just tested this on my end and was able to do so successfully. Is this only happening for you with that one issue or a specific user in your org/repo? Curious if you know whether other triagers are not experiencing this or if you're seeing this only in `/networkx.networkx`

As of right now those with `triage` permissions **cannot** do the conversions for issues in bulk via labels, correct.

**Comment by jarrodmillman** [comment]
Hi @becca !  Thanks for the response.  @MridulS could you check whether it is happening for you with just that one issue?  I will see if other team members are having this issue and let you know.  Just for a sanity check here is a screenshot of Mridul's permissions:
![permissions](https://user-images.githubusercontent.com/123428/102660526-04d16700-4130-11eb-89af-d8ec891255d1.png)
And here is a screenshot that Mridul sent me (**without**  the ``Convert to Discussion`` option):
![issue](https://user-images.githubusercontent.com/123428/102660847-98a33300-4130-11eb-98ab-c3ff0e8ff0d6.png)
And here is my screenshot as an organization owner (**with** the ``Convert to Discussion`` option):
![jarrod-issue](https://user-images.githubusercontent.com/123428/102661245-50d0db80-4131-11eb-8365-a1bc641828b0.png)
 
There are some other differences in the interface that we see.  For example, I can also edit the title.  I am not entirely sure what options should be available for members with triage permissions.  @becca is there anything else missing from Mridul's screenshot that you would expect to see for someone with triage permissions?

Obviously, we can work around this particular issue.  But the discussions seem very useful and we hope to grow the number of contributors with triage permissions.  So we would like to track this issue down.  Perhaps it is a browser issue?  Or maybe there is something odd about how I have configured our organization or repository?  Or maybe our organization has some legacy status and/or setting somewhere; we created the organization around 2010 (I believe).

**Reply by jarrodmillman** [reply]
@boothby you have triage permissions on the networkx/networkx repository.  Would you take a look at issue https://github.com/networkx/networkx/issues/4439 and let us know whether or not you have a ``Convert to discussion`` option in the lower right hand corner (like I do in the screenshot above)?

**Reply by boothby** [reply]
I do not have that option, no.  I tried logging out and logging back in again.

**Reply by evi-liu** [reply]
Can someone with `maintainer` or `admin` permissions confirm in the repository setting under `Manage access` that you indeed are currently under the `triage` role permission? There may be some oddities with team permissions.

**Reply by jarrodmillman** [reply]
Yes, this was an issue for people with ``triage`` role permissions.
![screenshot](https://user-images.githubusercontent.com/123428/105417577-adbc2780-5bf0-11eb-9055-a5906f8a3b68.png)

But at this point, I believe this feature has been disabled for everyone:
https://github.com/github/feedback/discussions/2924#discussioncomment-254592

**Reply by evi-liu** [reply]
Do you mind checking under the repository settings for these permissions? Repo -> Settings -> Manage access. It may be the case that these org settings aren't properly mapped to the repo settings. 

The feature has indeed been temporarily disabled, but this should be an independent problem.

**Reply by jarrodmillman** [reply]
Seems correct to me.
![screenshot](https://user-images.githubusercontent.com/123428/105619990-c6713c80-5dac-11eb-8780-41aa6ea1ee73.png)

**Comment by lesteve** [comment]
I reported a similar thing in https://github.com/github/feedback/discussions/3105#discussioncomment-295647 and after debugging further, it seems that:
- when triage rights are **granted through a team**, the "Convert to discussion" button does **not** appear
- when triage rights are **granted to an individual** the "Convert to discussion" button does appear

@evi-liu could it be a bug on the github side?

@jarrodmillman could you double-check that you see the same thing as me, i.e. that if you grant the triage rights to an individual (not through a team) the individual can see the "Convert to discussion" button?

For completeness, the repo I was debugging is https://github.com/joblib/test-convert-discussion-to-issue, granting @cmarmo the triage rights and testing whether she sees the "Convert to discussion" button in the issue https://github.com/joblib/test-convert-discussion-to-issue/issues/1.

**Reply by evi-liu** [reply]
yes, would appreciate if someone could check on their end to reproduce!

for what it's worth, i think this is likely the case, but would be good to double check before i report to the team :)

**Reply by lesteve** [reply]
I am confident that somehow triage rights granted through a team do not carry correctly for Github Discussions "Convert Issue to Discussion" feature so if you think this is likely the case maybe there is enough reason between the two of us to report it to your github team :wink:?

We originally saw the problem in https://github.com/scikit-learn/scikit-learn and I was able to debug and reproduce in https://github.com/joblib/test-convert-discussion-to-issue. Those are two separate organisations I would say this rules out some weird organisation settings.

In principle this should be quite easy to reproduce but let me know if there is something else I can do to help:
- create a repo **inside a organisation** and enable Github discussions on this repo
- grant triage permission **through a team** to an individual, let's call this person `i-am-a-triager` 
- create an issue, check that `i-am-a-triager` does NOT see the "Convert to discussion" button  

Optionally (this shows that individual triage permission does work):
- **grant triage permissions** to `i-am-a-triager` as an individual triager
- check that `i-am-a-triager` sees the "Convert to discussion" button

**Reply by jarrodmillman** [reply]
I can confirm that I see the exact same behavior on https://github.com/networkx/networkx
* when triage rights are **granted through a team**, the "Convert to discussion" button **does not** appear
* when triage rights are **granted to an individual**, the "Convert to discussion" button **does** appear

**Reply by lesteve** [reply]
@jarrodmillman thanks a lot for checking! Maybe consider editing your discussion title to better reflect the problem and let other people find it more easily?

So I guess this does confirm this is a bug on the github side ...

@evi-liu it would be great to have some kind of super approximate idea how long it may take to fix it. Less/more than a week would be precise enough as far as I am concerned.

If the bug is estimated to take more than a week I guess I'll give triage rights individually to our triagers, we don't have that many in scikit-learn (less than 5).

**Comment by lesteve** [comment]
@evi-liu I just double-checked and the bug is still there (Convert to discussion is not seen when triagers right is granted through a team). Could you provide some status on this:
- this is not yet clearly identified as a genuine bug (although it was reproduced in different projects)
- this is identified as a genuine bug, it is in your backlog somewhere, it will take some time but it will eventually be fixed
- some other status

I completely understand that the people working on Github Discussions are busy and that there are other higher priority issues to work on. In particular, there is an easy (but not very convenient) work-around for this bug : give triage permissions to individuals rather than a team.

Below is some honest feed-back that I will try not too make too harsh (let me know if I completely fail ...).

At the moment we (https://github.com/scikit-learn/scikit-learn, so a project of reasonable size, ~45k stars even if this is not a great metric by any means) are experimenting with Github Discussions to decide whether it seems like a good idea to help structuring the community around the project.

Seeing what naively seems like an easy to fix bug without much activity (or at least much visibility of the activity) around it kind of raises some doubts about the likelihood of what seems like harder functionalities to be implemented for example:
- Convert a Discussion to Issue github/feedback#2861
- Discussions labels github/feedback#19
